src/Qir/Samples/build-qir-samples.ps1

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+[CmdletBinding()]
+param (
+    [Parameter()]
+    [Switch]
+    $SkipQSharpBuild
+)
+
+& (Join-Path $PSScriptRoot .. .. .. build set-env.ps1)
+
+function Build-QirProject {
+    param (
+        [string]
+        $FolderPath
+    )
+
+    if (!$SkipQSharpBuild) {
+        Write-Host "##[info]Build Q# project for QIR samples '$FolderPath'"
+        dotnet build $FolderPath -c $Env:BUILD_CONFIGURATION -v $Env:BUILD_VERBOSITY
+        if ($LastExitCode -ne 0) {
+            Write-Host "##vso[task.logissue type=error;]Failed to compile Q# project at '$FolderPath' into QIR."
+            throw "Failed to compile Q# project at '$FolderPath' into QIR."
+        }
+    }
+}
+
+Write-Host "##[info]Compile Q# Projects into QIR"
+Build-QirProject (Join-Path $PSScriptRoot StandaloneInputReference qsharp)
+
+
+Write-Host "##[info]Build QIR samples"
+$oldCC = $env:CC
+$oldCXX = $env:CXX
+$oldRC = $env:RC
+
+$clangTidy = ""
+
+if (($IsMacOS) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin"))))
+{
+    Write-Host "On MacOS build QIR samples using the default C/C++ compiler (should be AppleClang)"
+}
+elseif (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
+{
+    Write-Host "On Linux build QIR samples using Clang"
+    $env:CC = "clang-11"
+    $env:CXX = "clang++-11"
+    $env:RC = "clang++-11"
+    $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy-11"
+}
+elseif (($IsWindows) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win"))))
+{
+    Write-Host "On Windows build QIR samples using Clang"
+    $env:CC = "clang.exe"
+    $env:CXX = "clang++.exe"
+    $env:RC = "clang++.exe"
+
+    if (!(Get-Command clang -ErrorAction SilentlyContinue) -and (choco find --idonly -l llvm) -contains "llvm") {
+        # LLVM was installed by Chocolatey, so add the install location to the path.
+        $env:PATH += ";$($env:SystemDrive)\Program Files\LLVM\bin"
+    }
+
+    if (Get-Command clang-tidy -ErrorAction SilentlyContinue) {
+        # Only run clang-tidy if it's installed. This is because the package used by chocolatey on
+        # the build pipeline doesn't include clang-tidy, so we allow skipping that there and let
+        # the Linux build catch tidy issues.
+        $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy"
+    }
+} else {
+    Write-Host "##vso[task.logissue type=error;]Failed to identify the OS. Will use default CXX compiler"
+}
```

Comment:
(Minor)
This looks more like a _warning_, rather than an _error_. But the fragment `type=error` makes me expect that it will be treated as an error.

Reply:
Good catch! This should be a warning, not an error.